cmd/metrics-server/app/options/kubelet_client.go

```diff
@@ -61,6 +63,9 @@ func (o *KubeletClientOptions) Validate() []error {
 	if (o.KubeletCAFile != "") && o.DeprecatedCompletelyInsecureKubelet {
 		errors = append(errors, fmt.Errorf("cannot use both --kubelet-certificate-authority and --deprecated-kubelet-completely-insecure"))
 	}
+	if metricResolution < o.KubeletRequestTimeout {
```

Comment:
In the previous comment I mentioned that we should have validation for **minimal** value of  `KubeletRequestTimeout`. For example `KubeletRequestTimeout` should be positive number as time.Duration can be negative. Setting timeout equal 0 also doesn't make sense.

Please revert the change to validate `metricResolution < o.KubeletRequestTimeout` in `KubeletClientOptions`. Kubelet client options should not know about the metricResolution and should validate only `KubeletRequestTimeout`.